docs/excel/excel-add-ins-pivots.md

```diff
@@ -0,0 +1,243 @@
+
+Data hierarchies are the values to be aggregated based on the row and column hierarchies. A PivotTable with a row hierarchy of **Farms** and a data hierarchy of **Crates Sold Wholesale** shows the sum total (by default) of all the different fruits for each farm.
+
+Filter hierarchies include or exclude data from the pivot based on values within that filtered type. A filter hierarchy of **Classification** with the type **Organic** selected only shows data for organic fruit.
+
+Here is the farm data again, alongside a PivotTable. The PivotTable is using **Farm** and **Type** as the row hierarchies, **Crates Sold at Farm** and **Crates Sold Wholesale** as the data hierarchies (with the default aggregation function of sum), and **Classification** as a filter hierarchy (with **Organic** selected). 
+
+![A selection of fruit sales data next to a PivotTable with row, data, and filter hierarchies.](../images/excel-pivot-table-and-data.png)
+
+This PivotTable could be generated through the JavaScript API or through the Excel UI. Both options allow for further manipulation through add-ins.
+
+## Create a PivotTable
+
+PivotTables need a name, source, and destination. The source can be a `Range`, `string`, or `Table`. The destination can either be a `Range` or `string`. 
+The following samples show various PivotTable creation techniques. All three samples feature a PivotTable named **Farm Sales** created on a worksheet called **PivotWorksheet** at cell **A2**. Its data comes from the worksheet **DataWorksheet** across the range **A1:E21**. 
+
+### Create a PivotTable with strings
+```ts
+await Excel.run(async (context) => {
+	context.workbook.worksheets.getActiveWorksheet()
+		.pivotTables.add("Farm Sales", "DataWorksheet!A1:E21", "PivotWorksheet!A2");
+
+	await context.sync();
+});
+```
+
+### Create a PivotTable with Range objects
+```ts
+await Excel.run(async (context) => {	
+	const myRange = context.workbook.worksheets.getItem("DataWorksheet").getRange("A1:E21");
+	const myRange2 = context.workbook.worksheets.getItem("PivotWorksheet").getRange("A2");
+	context.workbook.worksheets.getItem("PivotWorksheet").pivotTables.add("Farm Sales", myRange, myRange2);
+	
+	await context.sync();
+});
+```
+
+### Create a PivotTable at the workbook level
+```ts
+await Excel.run(async (context) => {
+	context.workbook.pivotTables.add("Farm Sales", "DataWorksheet!A1:E21", "PivotWorksheet!A2");
+
+	await context.sync();
+});
+```
+
+## Use an existing PivotTable
+Manually created PivotTables are also accessible through the PivotTable collection of the workbook or of individual worksheets. 
+The following code gets the first PivotTable in the worksheet. It then gives the table a name for easy future reference.
+
+```ts
+await Excel.run(async (context) => {
+	const pivotTableList = context.workbook.worksheets.getActiveWorksheet().pivotTables;
```

Comment:
lets do it from the workbook rather than from a worksheet
context.workbook.pivotTables.getItem('pivotName')